components/de_amp/browser/de_amp_url_loader.h

```diff
@@ -0,0 +1,137 @@
+  // loader.
+  static std::tuple<mojo::PendingRemote<network::mojom::URLLoader>,
+                    mojo::PendingReceiver<network::mojom::URLLoaderClient>,
+                    DeAmpURLLoader*>
+  CreateLoader(base::WeakPtr<DeAmpThrottle> throttle,
+               const GURL& response_url,
+               scoped_refptr<base::SingleThreadTaskRunner> task_runner,
+               DeAmpService* service,
+               content::WebContents* contents);
+
+ private:
+  DeAmpURLLoader(base::WeakPtr<DeAmpThrottle> throttle,
+                 const GURL& response_url,
+                 mojo::PendingRemote<network::mojom::URLLoaderClient>
+                     destination_url_loader_client,
+                 scoped_refptr<base::SingleThreadTaskRunner> task_runner,
+                 DeAmpService* service,
+                 content::WebContents* contents);
+
+  // network::mojom::URLLoaderClient implementation (called from the source of
+  // the response):
+  void OnReceiveEarlyHints(network::mojom::EarlyHintsPtr early_hints) override;
+  void OnReceiveResponse(
+      network::mojom::URLResponseHeadPtr response_head) override;
+  void OnReceiveRedirect(
+      const net::RedirectInfo& redirect_info,
+      network::mojom::URLResponseHeadPtr response_head) override;
+  void OnUploadProgress(int64_t current_position,
+                        int64_t total_size,
+                        OnUploadProgressCallback ack_callback) override;
+  void OnReceiveCachedMetadata(mojo_base::BigBuffer data) override;
+  void OnTransferSizeUpdated(int32_t transfer_size_diff) override;
+  void OnStartLoadingResponseBody(
+      mojo::ScopedDataPipeConsumerHandle body) override;
+  void OnComplete(const network::URLLoaderCompletionStatus& status) override;
+
+  // network::mojom::URLLoader implementation (called from the destination of
+  // the response):
+  void FollowRedirect(
+      const std::vector<std::string>& removed_headers,
+      const net::HttpRequestHeaders& modified_headers,
+      const net::HttpRequestHeaders& modified_cors_exempt_headers,
+      const absl::optional<GURL>& new_url) override;
+  void SetPriority(net::RequestPriority priority,
+                   int32_t intra_priority_value) override;
+  void PauseReadingBodyFromNet() override;
+  void ResumeReadingBodyFromNet() override;
+
+  void OnBodyReadable(MojoResult);
+  void OnBodyWritable(MojoResult);
+
+  void CompleteLoading(std::string body);
+  void CompleteSending();
+  void ForwardBodyToClient();
+  void SendReceivedBodyToClient();
+
+  void Abort();
+
+  base::WeakPtr<DeAmpThrottle> throttle_;
+
+  mojo::Receiver<network::mojom::URLLoaderClient> source_url_client_receiver_{
+      this};
+  mojo::Remote<network::mojom::URLLoader> source_url_loader_;
+  mojo::Remote<network::mojom::URLLoaderClient> destination_url_loader_client_;
+
+  GURL response_url_;
+  content::WebContents* contents_;
+  DeAmpService* de_amp_service;
+
+  scoped_refptr<base::SingleThreadTaskRunner> task_runner_;
```

Comment:
SingleThreadTaskRunner => SequencedTaskRunner ?

Reply:
I think I got that from Speedreader. Can change it to Sequenced, think that makes more sense.

Reply:
If you don't need to use a particular thread, just a sequence, please change.